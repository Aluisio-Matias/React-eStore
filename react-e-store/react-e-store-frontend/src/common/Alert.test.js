import { render } from "@testing-library/react";
import Alert from "../common/Alert";

test("it renders without crashing", () => {
    render(
        <Alert />
    );
});

test("matches snapshot for danger", () => {
    let messages = ["If everything is broken", "Try again"];
    const { asFragment } = render(<Alert type="danger" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot for success", () => {
    let messages = ["Everything is awesome now!"];
    const { asFragment } = render(<Alert type="success" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
});