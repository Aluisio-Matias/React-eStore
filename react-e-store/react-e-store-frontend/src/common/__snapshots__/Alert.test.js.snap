// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      If everything is broken
    </p>
    <p
      class="mb-0 small"
    >
      Try again
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Everything is awesome now!
    </p>
  </div>
</DocumentFragment>
`;
